src/hooks/useCheckAccess.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useRef } from 'react';
+import { useLocation, useNavigate } from 'react-router';
+
+import useClientIntegration from './useClientIntegration';
+
+import { addQueryParam, removeQueryParam } from '#src/utils/location';
+import { checkEntitlements, reloadActiveSubscription } from '#src/stores/AccountController';
+
+type intervalCheckAccessPayload = {
+  interval?: number;
+  iterations?: number;
+  offerId?: string;
+};
+const useCheckAccess = () => {
```

Comment:
This is a potential place to also add and remove the notification subscriptions so that you don't act on notifications that come delayed or out of order.

Reply:
Hm, that way the payment will take longer I guess, because time will be needed after calling the subscribe. 
I don't think there will be moments for delayed notifications since we already updated the subscribe to happen only when payment is submitted.